script.js

```
@@ -0,0 +1,252 @@
+let allTasks = JSON.parse(localStorage.getItem('allTasks')) || [];
+const host = 'http://localhost:8000';
+const hdrs = {
+  'Content-Type': 'application/json;charset=utf-8',
+      'Access-Control-Allow-Origin': '*'
+};
+
+window.onload = async () => {
+  const input = document.getElementById('addTask');
+  if (input === null) {
+    return;
+  }
+
+  input.focus();
+  input.addEventListener('change', updateValue);
+
+  const resp = await fetch(`${host}/allTasks`, {
+    method: 'GET'
+  });
+  const result = await resp.json();
+  allTasks = result;
+  localStorage.setItem('Tasks', JSON.stringify(allTasks));
+  render();
+}
+
+const addTask = async () => {
+  const input = document.getElementById('addTask');
+  if (input === null || input.value.trim() === '') {
+    return;
+  }
+
+  try {
+    const resp = await fetch(`${host}/createTask`, {
+      method: 'POST',
+      headers: hdrs,
+      body: JSON.stringify({
+        text: input.value,
+        isCheck: false,
+        creationTime: new Date()
+      })
+    });
+    const result = await resp.json();
+    allTasks = result;
+    localStorage.setItem('Tasks', JSON.stringify(allTasks));
+  }
+  catch(error) {
+    return;
```

here you need to throw an error to the client